python_modules/dagster/dagster/_core/instance/ref.py

```diff
@@ -111,26 +117,38 @@ def configurable_storage_data(
         )
 
     elif "sqlite" in config_field:
-        base_dir = check.str_elem(config_field["sqlite"], "base_dir")
+        base_dir = config_field["sqlite"]["base_dir"]
         storage_data = ConfigurableClassData(
             "dagster._core.storage.sqlite_storage",
             "DagsterSqliteStorage",
             yaml.dump({"base_dir": base_dir}, default_flow_style=False),
         )
-        run_storage_data = ConfigurableClassData(
-            "dagster._core.storage.runs",
-            "SqliteRunStorage",
-            yaml.dump({"base_dir": _runs_directory(base_dir)}, default_flow_style=False),
+        run_storage_data = (
+            ConfigurableClassData(
+                "dagster._core.storage.runs",
+                "SqliteRunStorage",
+                yaml.dump({"base_dir": _runs_directory(base_dir)}, default_flow_style=False),
+            )
+            if isinstance(base_dir, str)
+            else None
         )
-        event_storage_data = ConfigurableClassData(
-            "dagster._core.storage.event_log",
-            "SqliteEventLogStorage",
-            yaml.dump({"base_dir": _event_logs_directory(base_dir)}, default_flow_style=False),
+        event_storage_data = (
+            ConfigurableClassData(
+                "dagster._core.storage.event_log",
+                "SqliteEventLogStorage",
+                yaml.dump({"base_dir": _event_logs_directory(base_dir)}, default_flow_style=False),
+            )
+            if isinstance(base_dir, str)
+            else None
         )
-        schedule_storage_data = ConfigurableClassData(
-            "dagster._core.storage.schedules",
-            "SqliteScheduleStorage",
-            yaml.dump({"base_dir": _schedule_directory(base_dir)}, default_flow_style=False),
+        schedule_storage_data = (
+            ConfigurableClassData(
+                "dagster._core.storage.schedules",
+                "SqliteScheduleStorage",
+                yaml.dump({"base_dir": _schedule_directory(base_dir)}, default_flow_style=False),
+            )
+            if isinstance(base_dir, str)
+            else None
```

Comment:
i don't quite follow why we have to do this - I see the type set to `StringSource` on these inner classes